File: app/src/main/java/com/aaron/yespdf/main/IMainContract.java
package com.aaron.yespdf.main;

import android.content.Intent;

import androidx.annotation.StringRes;

import com.aaron.yespdf.common.IContract;

/**
 * @author Aaron aaronzzxup@gmail.com
 */
interface IMainContract {

    interface V extends IContract.V {
        void onShowMessage(@StringRes int stringId);

        void onShowLoading();

        void onHideLoading();

        void onUpdate();
    }

    interface P extends IContract.P {
        void insertPDF(Intent data);
    }
}

File: app/src/main/java/com/aaron/yespdf/main/IMainContract.kt
package com.aaron.yespdf.main

import androidx.annotation.StringRes
import com.aaron.yespdf.common.IPresenter
import com.aaron.yespdf.common.IView

/**
 * @author Aaron aaronzzxup@gmail.com
 */
interface IMainView : IView {
    fun onShowMessage(@StringRes stringId: Int)
    fun onShowLoading()
    fun onHideLoading()
    fun onUpdate()
}

abstract class IMainPresenter(view: IMainView) : IPresenter<IMainView>(view) {
    abstract fun insertPDF(paths: List<String>?, type: Int?, groupName: String?)
}